### ISSUE TITLE
[chart.js 2.0] Chart type not working

### ISSUE THREAD

**Issue by rafaellop** [jtblin/angular-chart.js#356]
Is there a way to change the chart type at runtime from e.g. line to bar?

**Comment by jtblin** [comment]
Yes use `chart-type`, it's in the readme and in the example site...

**Comment by rafaellop** [comment]
@jtblin Missed that because not listed with other directives, but still doesn't work:

`<canvas id="line" class="chart chart-base" chart-type="intakeChartType" ...`

`$scope.intakeChartType = 'line';`

Using that code with the latest `angular-chart.js` (alpha3) and `chart.js` causes error:

`Chart.controllers[type] is not a constructor
    at .<anonymous> (chart.min.js:3575)
    at Object.helpers.each (chart.min.js:4120)
    at buildOrUpdateControllers (chart.min.js:3564)
    at initialize (chart.min.js:3408)
    at new Chart.Controller 
`

**Comment by rafaellop** [comment]
@jtblin would you suggest some other way if this isn't working to change the chart type in runtime?

**Comment by fishgrind** [comment]
@rafaellop @jtblin I did a pr with my changes that fixed it for me. Have a look at https://github.com/fishgrind/angular-chart.js/pull/1/files

**Comment by jtblin** [comment]
@fishgrind I think you are not looking at the right branch, this is already done in chartjs-2.0 branch: https://github.com/jtblin/angular-chart.js/blob/chartjs-2.0/angular-chart.js#L169.

@rafaellop do you have a repro case?

**Comment by rafaellop** [comment]
@jtblin no, I don't have. It's a matter of just trying to use the following code:

```
<canvas id="line" class="chart chart-base" chart-data="data" chart-labels="labels" chart-type="type" ></canvas>
```

**Comment by jtblin** [comment]
Which angular-chart.js and which chart.js version?

**Comment by rafaellop** [comment]
I reported the error above with the latest angular-chart.js (alpha-3) and chart.js (2.0 dev) but I've just noticed there's a new dist of chart.js. I'll check this new dist and let know...

**Comment by fishgrind** [comment]
@jtblin Wow, that was stupid of me. You are right, I thought I was working with the latest files!

**Comment by rafaellop** [comment]
I've checked with the very latest files and the error still persists:

`Chart.controllers[type] is not a constructor
    at .<anonymous> (chart.js:3575)
    at Object.helpers.each (chart.js:4120)
    at buildOrUpdateControllers (chart.js:3564)
    at initialize (chart.js:3408)
    at new Chart.Controller (chart.js:3391)
    at new Chart (chart.js:5094)
    at createChart (angular-chart.js:169)
    at Object.fn (angular-chart.js:120)
    at n.$digest (ionic.bundle.js:166)
    at n.$apply (ionic.bundle.js:169)(anonymous function) @ ionic.bundle.js:142(anonymous function) @ ionic.bundle.js:114n.$digest @ ionic.bundle.js:166n.$apply @ ionic.bundle.js:169fn @ mn-touch.js:31onStartEvent @ mn-touch.js:81
`

Will try to do a repro.

**Comment by rafaellop** [comment]
This seems not to be an `angular-chart` error. The exception happens when `chart.js` tries to create a new controller for the new chart type. I suppose it's related to the Angular version bunded with my version of the ionic library.

**Comment by jtblin** [comment]
@fishgrind not so stupid ;) chart.js 2.0 has just been released and I haven't merged the 2.0 support branch on my side, planning to do that this weekend so at the moment there is a discrepancy between this repo and Chart.js latest version.

@rafaellop glad that you found the issue. It works for me on chart.js 2.0 and the 2.0 branch.
